### PR TITLE
chore(test/lambda): Add tests for lambda operations

### PR DESCRIPTION
**Pull request by shyamsfo**
chore(test/lambda): Add some mock tests for lambda operations (outline only)

### CODE REVIEW

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    LambdaFunctionProvider lambdaFunctionProvider = mock(LambdaFunctionProvider.class);
+    ReflectionTestUtils.setField(dlao, "lambdaFunctionProvider", lambdaFunctionProvider);
+
+    doReturn(lambdaClient).when(dlao).getLambdaClient();
+    doReturn(cachedFunction)
+        .when(lambdaFunctionProvider)
+        .getFunction(anyString(), anyString(), anyString());
+
+    InvokeRequest deleteRequest = new InvokeRequest();
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java:52]: deleteRequest?

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            .withCodeSha256("abc123def456");
+    doReturn(newLambdaInstance).when(lambdaClient).createFunction(createRequest);
+    // when
+    CreateFunctionResult output = clao.operate(null);
+    // then
+    verify(clao, atLeastOnce()).updateTaskStatus(anyString());
+    assertThat(output).isEqualTo(newLambdaInstance);
+  }
+
+  public static class PublishLambdaAtomicOperationTest {
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java:78]: this should prob be moved to it's own test file

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  private CreateFunctionResult mockInvokeResults() {
+    CreateFunctionResult result = new CreateFunctionResult().withFunctionName("abc");
+    return result;
+  }
+
+  @Test
+  void testPublishLambda_1() {
+    CreateFunctionResult answer = mockInvoker.operate(null);
+    assertThat(answer.getFunctionName()).isEqualTo("abc");
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java:45]: I'm not sure what this assertion is doing. It seems like we're completely mocking
```java
  public CreateFunctionResult operate(List priorOutputs) {
    updateTaskStatus("Initializing Creation of AWS Lambda Function Operation...");
    return createFunction();
  }
```

so it never gets called

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+public class CreateLambdaAtomicOperationTest {
+  private CreateLambdaAtomicOperation mockInvoker = mock(CreateLambdaAtomicOperation.class);
+  private CreateFunctionResult conv = mock(CreateFunctionResult.class);
+
+  {
+    when(mockInvoker.operate(any())).thenReturn(mockInvokeResults());
+  }
+
+  private CreateFunctionResult mockInvokeResults() {
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java:39]: seems like this was part of the old stuff, prob want to clean this guy up as well.

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/DeleteLambdaEventSourceAtomicOperationTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import com.amazonaws.services.lambda.model.DeleteEventSourceMappingRequest;
+import com.amazonaws.services.lambda.model.DeleteEventSourceMappingResult;
+import com.netflix.spinnaker.clouddriver.lambda.cache.model.LambdaFunction;
+import com.netflix.spinnaker.clouddriver.lambda.deploy.description.UpsertLambdaFunctionEventMappingDescription;
+import com.netflix.spinnaker.clouddriver.lambda.provider.view.LambdaFunctionProvider;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.util.ReflectionTestUtils;
+
+public class DeleteLambdaEventSourceAtomicOperationTest implements LambdaTestingDefaults {
+  private DeleteLambdaEventSourceAtomicOperation mockDeleter =
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/DeleteLambdaEventSourceAtomicOperationTest.java:33]: seems like this isn't used

**Review by kevinawoo — CHANGES_REQUESTED** [review]

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    invokeDesc.setFunctionName(fName).setQualifier(version).setRegion(region).setAccount(account);
+
+    InvokeLambdaAtomicOperation invokeOperation = spy(new InvokeLambdaAtomicOperation(invokeDesc));
+    doNothing().when(invokeOperation).updateTaskStatus(anyString());
+
+    AWSLambda lambdaClient = mock(AWSLambda.class);
+    LambdaFunction cachedFunction = getMockedFunctionDefintion();
+
+    LambdaFunctionProvider lambdaFunctionProvider = mock(LambdaFunctionProvider.class);
+    ReflectionTestUtils.setField(invokeOperation, "lambdaFunctionProvider", lambdaFunctionProvider);
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java:45]: 👏 nice!

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/LambdaTestingDefaults.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+package com.netflix.spinnaker.clouddriver.lambda.deploy.ops;
+
+import com.amazonaws.services.lambda.model.AliasConfiguration;
+import com.amazonaws.services.lambda.model.EventSourceMappingConfiguration;
+import com.netflix.spinnaker.clouddriver.lambda.cache.model.LambdaFunction;
+import java.util.ArrayList;
+import java.util.List;
+
+public interface LambdaTestingDefaults {
```
Comment (kevinawoo) [inline clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/LambdaTestingDefaults.java:25]: 👏  nice! 💯

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java (hunk 1 above)
Comment (shyamsfo) [reply clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/InvokeLambdaAtomicOperationTest.java]: Updated the name here and all other files

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java (hunk 2 above)
Comment (shyamsfo) [reply clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java]: Done

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java (hunk 3 above)
Comment (shyamsfo) [reply clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java]: Removed this dummy function.

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java (hunk 4 above)
Comment (shyamsfo) [reply clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/CreateLambdaAtomicOperationTest.java:39]: Done

File: clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/DeleteLambdaEventSourceAtomicOperationTest.java (hunk 5 above)
Comment (shyamsfo) [reply clouddriver-lambda/src/test/java/com/netflix/spinnaker/clouddriver/lambda/deploy/ops/DeleteLambdaEventSourceAtomicOperationTest.java]: Removed

**Review by kevinawoo — APPROVED** [review]

**Pull request merged**